holoviews/element/path.py

```diff
@@ -49,18 +49,18 @@ class Path(Geometry):
 
     group = param.String(default="Path", constant=True)
 
-    datatype = param.ObjectSelector(default=['multitabular'])
+    datatype = param.ObjectSelector(default=['multitabular']+datatypes)
 
     def __init__(self, data, kdims=None, vdims=None, **params):
         if isinstance(data, tuple) and len(data) == 2:
             x, y = map(np.asarray, data)
-            if y.ndim == 1:
-                y = np.atleast_2d(y).T
-            if len(x) != y.shape[0]:
-                raise ValueError("Path x and y values must be the same length.")
-            data = [np.column_stack((x, y[:, i])) for i in range(y.shape[1])]
+            if y.ndim > 1:
+                if len(x) != y.shape[0]:
+                    raise ValueError("Path x and y values must be the same length.")
+                data = [np.column_stack((x, y[:, i])) for i in range(y.shape[1])]
         elif isinstance(data, list) and all(isinstance(path, Path) for path in data):
-            data = [p for path in data for p in path.data]
+            data = [p for path in data for p in
+                    (path.data if isinstance(path.data, list) else [path.data])]
```

Comment:
This might be clearer as:

```
path_data = path.data if isinstance(path.data, list) else [path.data]
data = [p for path in data for p in path_data]
```
But it's still hard to read all the "path"s and "data"s here....

Reply:
That's not going to work, we're iterating over the items in each ``path.data``, i.e. you're iterating over the inner loop here which isn't defined by that point.